language/bytecode-verifier/src/dependencies.rs

```diff
@@ -88,13 +91,21 @@ impl<'a, 'b> Context<'a, 'b> {
                     (module_id.clone(), func_name.to_owned()),
                     func_def.visibility,
                 );
-                match func_def.visibility {
-                    Visibility::Private => continue,
-                    Visibility::Public | Visibility::Script => {
-                        context
-                            .func_id_to_handle_map
-                            .insert((module_id.clone(), func_name.to_owned()), func_def.function);
+                let may_be_called = match func_def.visibility {
+                    Visibility::Public | Visibility::Script => true,
+                    Visibility::Friend => {
+                        if let Some(self_module_id) = &self_module {
+                            friend_module_ids.contains(self_module_id)
+                        } else {
+                            false
+                        }
```

Comment:
if-let + else is not my friend :P 

Either switch to a match, or `self_module.map(|self_id| friend_module_ids.contains(self_id)).unwrap_or(false)`